test/integration/discovered_hosts_test.rb

```diff
@@ -29,24 +31,70 @@ class DiscoveredHostsTest < IntegrationTestWithJavascript
       assert page.has_text?('The following hosts are about to be changed')
       page.find_button('Submit').click
     end
+
+    test 'triggers reboot on all discovered_hosts' do
+      Host::Discovered.any_instance.stubs(:subnet).returns(subnet)
+      Host::Discovered.any_instance
+                      .expects(:reboot)
+                      .at_least(discovered_hosts.count)
+      select_all_hosts
+      page.find_link('Select Action').click
+      page.find_link('Reboot').click
+      wait_for_ajax
+      assert page.has_text?('The following hosts are about to be changed')
+      page.find_button('Submit').click
+    end
+
+    test 'shows warning for all hosts with missing subnet' do
+      select_all_hosts
+      page.find_link('Select Action').click
+      page.find_link('Auto Provision').click
+      wait_for_ajax
+      assert page.has_text?('The following hosts are about to be changed')
+      page.find_button('Submit').click
+      wait_for_ajax
+      assert page.has_text?("Discovered hosts reported from unknown subnet")
+    end
   end
 
-  describe 'Multiple host Autoprovision' do
-    test 'converts all discovered to managed hosts' do
+  describe 'Perform host Autoprovision' do
+    test 'converts single discovered to managed host' do
+      Host::Discovered.any_instance.stubs(:subnet).returns(subnet)
       select_host_checkbox(discovered_host.id)
       page.find_link('Select Action').click
       page.find_link('Auto Provision').click
       wait_for_ajax
       assert page.has_text?('The following hosts are about to be changed')
       page.find_button('Submit').click
+    end
+
+    test 'converts all discovered to managed hosts' do
+      Host::Discovered.any_instance.stubs(:subnet).returns(subnet)
+      select_all_hosts
+      page.find_link('Select Action').click
+      page.find_link('Auto Provision').click
+      wait_for_ajax
+      assert page.has_text?('The following hosts are about to be changed')
+      page.find_button('Submit').click
       wait_for_ajax
       assert page.has_text?('Discovered hosts are provisioning now')
     end
+
+    test 'shows warning for all hosts with missing subnet' do
+      select_all_hosts
+      page.find_link('Select Action').click
+      page.find_link('Auto Provision').click
+      wait_for_ajax
+      assert page.has_text?('The following hosts are about to be changed')
+      page.find_button('Submit').click
+      wait_for_ajax
+      assert page.has_text?("Discovered hosts reported from unknown subnet")
+    end
   end
 
   describe 'Delete hosts' do
     test 'it removes all hosts' do
-      select_host_checkbox(discovered_host.id)
+      select_all_hosts
```

Comment:
The `select_all_hosts` private method was not being used anywhere, therefore I created 2 test cases:
1) single host (reboot, autoprovision) which uses the s`elect_host_checkbox` method.
2) Multiple hosts (reboot, auto provision and delete) which uses the `select_all_hosts` method.